File: neetcode/167. Two Sum II - Input Array Is Sorted/167.java
/*
Method 1.0: Two pointers
TC: O(N)
SC: O(1)
 */

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.length - 1;

        while (left < right) {
            int currTotal = numbers[left] + numbers[right];
            if (currTotal == target) {
                return new int[]{left + 1, right + 1};
            }

            else if (currTotal < target) {
                left++;
            } else {
                right--;
            }
        }

        return new int[]{-1, -1};
    }
}

File: neetcode/167. Two Sum II - Input Array Is Sorted/167.kt
/*
Method 1.0: Two pointers
TC: O(N)
SC: O(1)
 */

class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        var left = 0
        var right = numbers.size - 1

        while (left < right) {
            val currTotal = numbers[left] + numbers[right]

            if (currTotal == target) {
                return intArrayOf(left + 1, right + 1)
            } else if (currTotal < target) {
                left++
            } else {
                right--
            }
        }

        return intArrayOf(-1, -1)
    }
}
